tests/FakeItEasy.Specs/OrderedCallMatchingSpecs.cs

```diff
@@ -54,29 +85,61 @@ public interface IHaveOneGenericParameter
-                        using (var scope = Fake.CreateScope())
-                        {
-                            fake.Bar(1);
-                            fake.Bar(new Generic<bool>());
-                            using (scope.OrderedAssertions())
-                            {
-                                A.CallTo(() => fake.Bar(A<Generic<bool>>.Ignored)).MustHaveHappened();
-                                exception = Record.Exception(() => A.CallTo(() => fake.Bar(A<int>.Ignored)).MustHaveHappened());
-                            }
-                        }
-                    });
-
-            "it should tell us that the call was not matched"
+            "And a call on the Fake, passing argument of type int"
+                .x(() => fake.Bar(1));
+
+            "And a call on the Fake, passing argument of type Generic<bool>"
+                .x(() => fake.Bar(new Generic<bool>()));
+
+            "And a call-ordering context"
+                .x(c => sequentialCallContext = A.SequentialCallContext());
+
+            "When I expect the call with any argument of type int to have been made"
+                .x(() => A.CallTo(() => fake.Bar(A<int>.Ignored)).MustHaveHappened().InOrder(sequentialCallContext));
+
+            "And I expect the call with any argument of type Generic<bool> to have been made next"
+                .x(() => exception = Record.Exception(() => A.CallTo(() => fake.Bar(A<Generic<bool>>.Ignored)).MustHaveHappened().InOrder(sequentialCallContext)));
+
+            "Then the assertion should succeed"
+                .x(() => exception.Should().BeNull("because the assertion should have succeeded"));
+        }
+
+        [Scenario]
+        public static void GenericCallsFailure(
+            IHaveOneGenericParameter fake,
+            ISequentialCallContext sequentialCallContext,
+            Exception exception)
+        {
+            "Given a Fake"
+                .x(() => fake = A.Fake<IHaveOneGenericParameter>());
+
+            "And a call on the Fake, passing argument of type int"
+                .x(() => fake.Bar(1));
+
+            "And a call on the Fake, passing argument of type Generic<bool>"
+                .x(() => fake.Bar(new Generic<bool>()));
+
+            "And a call-ordering context"
+                .x(c => sequentialCallContext = A.SequentialCallContext());
+
+            "When I expect the call with any argument of type Generic<bool> to have been made"
+                .x(() => A.CallTo(() => fake.Bar(A<Generic<bool>>.Ignored)).MustHaveHappened().InOrder(sequentialCallContext));
+
+            "And I expect the call with any argument of type int to have been made next"
+                .x(() => exception = Record.Exception(() => A.CallTo(() => fake.Bar(A<int>.Ignored)).MustHaveHappened().InOrder(sequentialCallContext)));
+
+            "Then the assertion should fail"
+                .x(() => exception.Should().NotBeNull("because the assertion should have failed"));

```

Comment:
An alternative would be to check that the exception `BeAnExceptionOfType`, if you want to be more specific about it.